src/index.ts

```diff
@@ -394,12 +394,18 @@ export function Graph(serialized?: Serialized) {
   function shortestPaths(source: NodeId, destination: NodeId) {
     let path = shortestPath(source, destination);
     const paths = [path],
-      removedEdges = [],
+      removedEdges: any = [],
       weight = path.weight;
     while (weight) {
-      removeEdge(path[0], path[1]);
-      removeEdge(path[1], path[0]);
-      removedEdges.push([path[0], path[1]]);
+      [
```

Comment:
This code creates many new arrays and one new closure each time it runs, which could degrade performance as all of those things need to be garbage collected.

I would suggest the following refactoring:

```
    const u = path[0];
    const v = path[1];

    if (hasEdge(u, v)) {
        removedEdges.push({ u, v, weight: getEdgeWeight(u, v) });
        removeEdge(u, v);
    }

    if (hasEdge(v, u)) {
        removedEdges.push({ u: v, v: u, weight: getEdgeWeight(v, u) });
        removeEdge(v, u);
    }
```